x/programs/rust/README.md

```diff
@@ -25,6 +25,100 @@ Just a couple things to note. Serialization is minimal, yet there are certain as
 
 On the Go side, we unmarshal in the same order.
 
+## Storage
+
+A Program has 2 storage types.
+
+- `Memory` in WebAssembly is a linear buffer of unsigned bytes that can read and
+written to by both the guest or host. Because of the very minimal types
+supported by WebAssembly Memory is critical for communication between `Program`
+and host.
+
+- `State` is a host module which is imported into the `Program` instance, which
+exposes access to a database store. Unlike `Memory` state can be durable.
+
+### Memory
+
+In this example we share a byte slice between our host and a `Program`. To
+achieve this we can write our byte slice to the shared memory from he host using
```

Comment:
```suggestion
achieve this we can write our byte slice to the shared memory from the host using
```

Reply:
Also, "To achieve this," (missing comma)